builtins: add ST_<Shape>FromWKB and ST_<Shape>FromText functions

otan opened the pull request:
Release note (sql change): Added builtins allowing parsing text and wkb
all supported shapes, e.g. `ST_PointFromWKB`, `ST_PointFromText`, which
return NULL if the WKT/EWKT/WKB do not match the shape provided.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/48441)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48441#-:-M6fM3xI7vZlaD7hcRwG:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/sql/sem/builtins/geo_builtins.go, line 320 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/48441#-M6fL8-b6yGsnTBnHmYt:-M6fL8-b6yGsnTBnHmYu:b-4edpyd) ([raw file](https://github.com/cockroachdb/cockroach/blob/1494ed19adf158c7ba151abd77f9d4152b245d06/pkg/sql/sem/builtins/geo_builtins.go#L320)):*
> ```Go
> 	"st_multilinestringfromtext": geometryFromTextCheckShapeBuiltin(geopb.Shape_MultiLineString),
> 	"st_multilinestringfromwkb":  geometryFromWKBCheckShapeBuiltin(geopb.Shape_MultiLineString), // missing from PostGIS
> 	"st_multipointfromtext":      geometryFromTextCheckShapeBuiltin(geopb.Shape_MultiPoint),     // SRID version missing form PostGIS
> ```

s/form/from/


<!-- Sent from Reviewable.io -->

otan committed:
builtins: add ST_<Shape>FromWKB and ST_<Shape>FromText functions

Release note (sql change): Added builtins allowing parsing text and wkb
all supported shapes, e.g. `ST_PointFromWKB`, `ST_PointFromText`, which
return NULL if the WKT/EWKT/WKB do not match the shape provided.

otan commented:
bors r=sumeerbhola
